schema/types/volume_test.go

```diff
@@ -0,0 +1,40 @@
+package types
+
+import (
+	"testing"
+)
+
+func TestVolumeFromString(t *testing.T) {
+	tests := []string{
+		"foobar,kind=host,source=/tmp",
+		"foobar,kind=host,source=/tmp,readOnly=true",
+		"foobar,kind=empty",
+		"foobar,kind=empty,readOnly=true",
+	}
+	for i, in := range tests {
+		l, err := VolumeFromString(in)
+		if err != nil {
+			t.Errorf("#%d: got err=%v, want nil", i, err)
+		}
+		if l == nil {
```

Comment:
really this should DeepEqual the resulting Volume, checking non-nil is lazy :stuck_out_tongue: